### ISSUE TITLE
Use Storybook Controls addon to enable more interactive stories.

### ISSUE THREAD

**Issue by techanvil** [google/site-kit-wp#6351]
## Feature Description

Storybook has a Controls addon that allows UI controls to be generated from Storybook `args`.

This is useful for allowing interactive updates of component props and so forth, making stories more flexible and useful for trying out components in various permutations. This should also reduce the number of story variants we need to manually code.

We should integrate the Controls addon. As we already use `args` some careful attention will need to be made to ensure existing stories still work as expected and without creating unwanted UI controls, potentially by moving away from `args` for existing stories.

We could follow up by reworking some of our existing stories to reduce duplication.

See https://storybook.js.org/docs/6.3/react/essentials/controls

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by aaemnnosttv** [comment]
I agree that adding support for controls increases the utility of our stories, I don't think that it necessarily includes a substantial opportunity to reduce the number of stories we have. Especially any stories that are enabled for VRT, these need to stay.

Generally speaking, a story should be cheap to define, and we shouldn't worry too much about how many we have (the same cannot be said for those with VRT enabled, which need more consideration now).

The "weight" of stories beyond VRT is in the setup which is needed to put the datastore in a specific state, which I don't think can be solved with controls.

With that said, again, I am supportive of adding controls, they may just not be very useful for our mostly "connected" components which don't use `args`, but are based on data store state.
